### DISCUSSION TITLE
How to get Claude working?

### DISCUSSION THREAD

**Discussion by benwoodward** [zed-industries/zed#10294] [Feature Requests]
I'm using the preview version of Zed (v0.130.1, b33f7c64f6b2567aa53de22c7191dd15e7843245), I've reset my API key, and input my Antropic key, added the below to my settings, and selected Claude Opus from the as the model in the assistant panel, and I get an error: "RPC CompleteWithLanguageModel failed - permission denied"

```
  "assistant": {
    "dock": "left",
    "version": "1",
    "provider": {
      "name": "zed.dev"
    }
  }
```

What am I doing wrong?

![image](https://github.com/zed-industries/zed/assets/1472981/f13cfde8-c733-4425-ac57-b14e801d880e)

**Comment by JosephTLyons** [comment]
Hey @benwoodward, you aren't doing anything wrong here, it is just that this feature (using zed.dev as a provider) is not ready for public consumption and is behind a feature flag at the moment. We should be opening it up here shortly. That being said, we should do a better job of communicating when a feature is behind a flag - sorry about that.

**Reply by benwoodward** [reply]
ok, great thanks.

**Reply by hansstimer** [reply]
Hey @JosephTLyons, would like to be able to use Claude Sonnet 3.5, if the Assistant feature can pull that off yet.

**Reply by sguestsmith** [reply]
I'll second this, would like to pull in Claude if it's now available?

**Comment by heyAyushh** [comment]
This will be helpful

**Comment by ChristianPerez34** [comment]
I was able to get claude working but i noticed zed doesn't have the 3.5 sonnet model. Is this something that will be implemented in the near future? 

<img width="199" alt="image" src="https://github.com/zed-industries/zed/assets/22486123/1289c300-8f28-4cfe-8b7a-1a12939db494">

**Reply by pinkavaj** [reply]
How did you make it working, can you geve a hint pls?

**Reply by pinkavaj** [reply]
This might be helpfull https://github.com/zed-industries/zed/issues/8914
